### ISSUE TITLE
Prepare dev-env for ARENA

### ISSUE THREAD

**Issue by IgnacioHeredia** [ai4os/ai4os-dev-env#3]
I spoke with Álvaro that we should prepare the devenv for the ARENA era. From the discussion, two items emerged:

1. we should streamline the usage of [AI4EOSC models as code assistants](https://docs.ai4eosc.eu/en/latest/reference/llm.html#use-it-as-a-code-assistant-with-vscode). The thing is that Continue [is discontinued](https://continue.dev/) (pun intended), as it got acquired by Cursor. So we should:

   * select a new extension to replace continue. My initial candidates where:
      - Copilot: my initial preferred option as it is very well integrated with VSCode and support OpenAI compatible endpoints. Bad reviews mainly come from the pricing change in Github provided models.  
      - Cline: does not have great review lately.
      - Kilocode: mixed reviews

      But I'm sure other options exist.

    * modify deepstart (or the built images to have the extension installed by default)
    * investigate what configuration (endpoint, tokens, etc)  is needed and where should it be be saved, to have it ready to go without any user action (configuration can be injected by PAPI)  

2. maybe we should add a coding agent independent from VSCode or Jupyter. 
  
   - It could be an agent with a dedicated IDE like [OpenCode](https://opencode.ai/), which could be deployed in a new port
   - It could be a terminal-based agent (like Pi, Kilocode, aider, etc).

In my view, item **1** would have higher priority since I guess most people use VScode to develop.

Since you are updating the dev-env, could you take a look into it? Or if you are too busy, I can try to take a look at it when I come back from holidays, and do a PR.

**Comment by vykozlov** [comment]
Hi @IgnacioHeredia , @alvarolopez 

somehow github messages get buried among my other emails :-(
I am actively working on ai4os-dev-env. What is available now:
1. VScode is pre-installed in all (new) images (jupyterlab too, as before)
2. OpenCode is also already pre-installed in all (new) images

Additional JupyterLab, VSCode, (potentially OpenCode) plugins/extensions are installed via [deep-start](https://github.com/ai4os/deep-start). New images already pull deep-start from the `v3.0.0` branch (see below).

deep-start workflow (preparing as release 3.0.0 (currently branch v3.0.0) :
1. ✔️  if OpenCode is not installed - install it
2. ✔️  OpenCode comes with [opencode.jsonc](https://github.com/ai4os/deep-start/blob/64273ac301f52e8ed20916648dcffc9ba6502076/opencode/opencode.jsonc) file pre-configured for AI4EOSC LLM. Pre-configured currently:
    * AI4EOSC/default
    * AI4EOSC/Qwen/Qwen3-14B
    * apiKey can be automatically injected if `{env:AI4EOSC_LLM_KEY}` is provided
 3. ✔️ Generally speaking, in order to use OpenCode within either JupyterLab or VScode, we don't need a new port - OpenCode can work from the terminal window. BUT it uses so-called TUI interface which often has a problem with copy/paste. It is nevertheless, sort-of solved:
    * JupyterLab has extension [jupyterlab-terminal-clipboard-extension](https://github.com/stellarshenson/jupyterlab_terminal_clipboard_extension) that solves copy/paste problem
    * OpenCode has plugin [OpenChamber](https://open-vsx.org/extension/FedaykinDev/openchamber) to interface with OpenCode in the VScode editor directly
 5. ✔️  OpenCode itself implemented `opencode serve` mode with BasicAuth, i.e. there is already default OpenCode web interface, thus we can start OpenCode in web-browser very similar way as JupyterLab or VScode. `deep-start`  has now corresponding implementation: `deep-start -o` or `deep-start --opencode` . Meaning, a user can provide a password via the dashboard, `deep-start --opencode` will serve OpenCode desktop on the usual IDE Port. The only difference, OpenCode web interface also asks for Username, which defaults to `opencode`, but the password is the one user configures during deployment.
 6. ✔️  Continue.dev is also listed among VScode extentsions to be installed by `deep-start `and it is pre-configured for AI4EOSC LLMs (see [config.yml](https://github.com/ai4os/deep-start/tree/64273ac301f52e8ed20916648dcffc9ba6502076/continue/config.yaml)). If `AI4EOSC_LLM_KEY` environemnt is present, during launch of the container  `deep-start` will replace it in the `config.yaml`. However, indeed, it seems Continue.dev is deprecated :-(((
 7. ✔️  JupyterLab and VScode extensions are now listed in one configuration file [ide-extensions.ini](https://github.com/ai4os/deep-start/blob/64273ac301f52e8ed20916648dcffc9ba6502076/ide-extensions.ini).
Having it dynamically loaded has the advantage that we don't need to re-build Docker images just for one more (or less) extension. Docker image build is expensive at Jenkins. Feedback to the list of extensions is welcome :-)
8.  ✔️ On the dashboard the following new images are already available, that include items 1-7 above and based on:
     * ubuntu:24.04
     * cuda13.0.3, cuda12.8.1, cuda12.6.3, cuda12.4.1
     * pytorch 2.3, 2.4, 2.6
     * tensorflow 2.14, 2.16.2, 2.18.0
9. To-Do: I am considering to implement auto-update of deep-start. There is already a flag `deep-start -i` but I want to think more. The current idea: restructure git repository, make auto-update mandatory but safe: try the auto-update, if fails, stay with the version already available in the Docker image.
10. ToDo: I am also considering that configuration files of e.g. OpenCode / other tools should be dynamically loaded. Though depends on the naming policy of LLMs at AI4EOSC - we may fix the names: e.g. `default` (already), `large`, `medium`, `small`, `coder`, `embedding` etc (our local KIT LLM service is going to introduce something like that). Otherwise the configs available in Docker images might easily become outdated.
11. To-Do: build more images. Latest pytorch is 2.12, tensorflow 2.21
12. To-Do: implement marking of older builds as `-legacy`, i.e. `ai4os-dev-env:u20.04-legacy` indicating that they are neither updated nor maintained anymore. By `-legacy` word may also filter them on the dashboard. Some people might still want them but we make it clear - no maintenance.
13. some other ideas :-)

**Comment by IgnacioHeredia** [comment]
> somehow github messages get buried among my other emails :-(

haha I have a dedicated email filter and folder for them to avoid that :P

Excellent Valentin, looks great! 🚀 

Some comments:

> apiKey can be automatically injected if {env:AI4EOSC_LLM_KEY} is provided

Great, I'll look into injecting in from PAPI. But since we currently do not save LLM API keys, I'll need to add some logic to save/retrieve them from Vault and make sure Vault is synced with LiteLLM. I'll try to coordinate with Marta which is doing something similar for the new GenAI UI.

> we can start OpenCode in web-browser very similar way as JupyterLab or VScode

Great, although since (from the screenshots I saw) you do not seem to have direct access to your code from Opencode (only to the agent terminal), I'm not sure about the usefulness of deploying Opencode in standalone. I saw it more as a complement to VScode or JupyterLab. But if you see the value in having it standalone, i can add the option to PAPI/Dashboard.

> Feedback to the list of extensions is welcome :-)

Here is a subset of the ones I use that I don't see in the list and that might be relevant for a typical user, in case any of them sounds appealing to you:

  - [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker),
  - [Todo highlight](https://marketplace.visualstudio.com/items?itemName=wayou.vscode-todo-highlight) 
  - [Python Indent](https://marketplace.visualstudio.com/items?itemName=KevinRose.vsc-python-indent)
  - [Ruff](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff) --> faster than black, yet compatible
  - [Github Pull Requests](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github)
  - [CSV](https://marketplace.visualstudio.com/items?itemName=ReprEng.csv&ssr=false#overview) (jonaraphael)
  - [Git Graph](https://marketplace.visualstudio.com/items?itemName=mhutchie.git-graph) 
  - [Git Tree Compare](https://marketplace.visualstudio.com/items?itemName=letmaik.git-tree-compare) - to compare between commits
  - [Even Better TOML](https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml&ssr=false#review-details)

> Though depends on the naming policy of LLMs at AI4EOSC - we may fix the names: e.g. default (already), ...

Yes. To avoid having to constantly [update the docs](https://docs.ai4eosc.eu/en/latest/reference/llm.html#implement-a-rag-pipeline) with new models, we recently introduced model aliases:
* `default`: for a general purpose model (currently points to mistral)
* `default-embeddings`: for embeddings models (currently points to qwen 4B)

We might introduce more in the future (e.g. `default-agent`, etc)  but I rather wait to see which ones we really need to avoid having to roll back some of them.

> implement marking of older builds as -legacy

Legacy label sound great.

> some other ideas :-)

Now that you are opening the can of worms, I always wondered why the user code is located in `/srv` instead of `/home`. Gemini tells me `/home` make more sense but I might be getting sycophanted. :P

Thanks!

**Comment by IgnacioHeredia** [comment]
Next week I'll try the new images and provide some feedback if any. 👍🏽

**Comment by IgnacioHeredia** [comment]
Hi @vykozlov ,

`AI4EOSC_LLM_KEY` is now automatically injected in devenv and `opencode` is now an option to deploy (along vscode and jupyter) (https://github.com/ai4os/ai4-papi/commit/db3fa4917a43e0e028d192c16bf626ca9e513964). Dashboard still needs to be adapted for opencode option (https://github.com/ai4os/ai4-dashboard/issues/528).

**Feedback:**
I tested both  vscode with openchamber and opencode, landing in PNSC.
Main comments:

* While AI4EOSC models do appear in the UI, trying to use them raises this error:

<img width="399" height="465" alt="Image" src="https://github.com/user-attachments/assets/f6a408a9-6ff8-4285-bd08-b4b81af39a38" />

The current LiteLLM rate limit we have for `ap-d` is 30 requests per minute and 30K tokens per minute, which is the higher tier of all ap teams. I can increase this limit further if needed but I would like to understand first why openchamber reaches the limit with a "how are you" question.

In the case of opencode, the UI does not even show an error message (unlike openchamber), it just doesn't reply.

* I'm not sure about how AI4EOSC models are included:

<img width="527" height="248" alt="Image" src="https://github.com/user-attachments/assets/0e44fbc3-117c-49f6-a1ed-65251c563c19" />

If hardcoded, then I would only hardcode default ones (`default` and `default-embeddings`).
If added by automated discovery, then many models are missing.
